File: algorithm/src/test/java/programmers/level4/사단_고음/SolutionTest.java
package programmers.level4.사단_고음;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

import static org.assertj.core.api.Assertions.assertThat;


public class SolutionTest {

    @ParameterizedTest
    @CsvSource({
            "15, 1",
            "24, 0",
            "41, 2",
            "2147483647, 1735",
    })
    void solutionTest(int n, int answer) {
        Solution solution = new Solution();

        assertThat(solution.solution(n)).isEqualTo(answer);
    }
}

File: algorithm/src/test/kotlin/io/yomni/programmers/level1/id12925/SolutionTest.kt
package io.yomni.programmers.level1.id12925

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource

@DisplayName("SolutionTest")
class SolutionTest {
    companion object {
        @JvmStatic
        fun argumentsProvider() = listOf(
            Arguments.of("1234", 1234),
            Arguments.of("-1234", -1234),
        )
    }

    @ParameterizedTest
    @MethodSource("argumentsProvider")
    fun solution(input: String, expected: Int) {
        val solution = Solution()
        assertThat(solution.solution(input)).isEqualTo(expected)
    }
}
